### PR TITLE
Nashawn/ch22/add follow and unfollow functionality

### PR DESCRIPTION
**Pull request by nashawn-griffith**
I closed the other PRS for this feature. 
You can use this PR.

Feat:
Added follow and unfollow functionality.
The logged in user is able to see the posts of all users he/she is following
-Modifications were made to the mutations and queries to add this functionality

### CODE REVIEW

File: server/schema/mutations.js [hunk 1]
```diff
@@ -71,12 +71,18 @@ const mutation = new GraphQLObjectType({
         body: {type: GraphQLString},
         tags: {type: GraphQLList(GraphQLString)}
       },
-      resolve(parentValue, {title, body, tags}, context) {
-        if (!context.user) {
```
Comment (kpjackson90) [inline server/schema/mutations.js]: why did you remove context here?

File: server/schema/mutations.js [hunk 2]
```diff
@@ -185,6 +191,23 @@ const mutation = new GraphQLObjectType({
...
         return Post.remove({_id: id});
       }
+    },
+    followUser: {
+      type: UserType,
+      args: {
+        userId: {type: new GraphQLNonNull(GraphQLID)}
+      },
+      async resolve(parentValue, {userId}, {user}) {
+        if (!user) {
```
Comment (kpjackson90) [inline server/schema/mutations.js]: how come we are not using context any more? @nashawn-griffith

File: server/schema/mutations.js (hunk 1 above)
Comment (nashawn-griffith) [reply server/schema/mutations.js]: At the time of writing the code i thought it would be easier to understand

File: server/schema/mutations.js (hunk 2 above)
Comment (nashawn-griffith) [reply server/schema/mutations.js]: Same reason as above

File: server/schema/mutations.js (hunk 1 above)
Comment (kpjackson90) [reply server/schema/mutations.js]: @nashawn-griffith context is a graphql standard thing for authentication. I see that you didn't change it in the server.js file, so did you destructure user from context somewhere? Am not seeing that anywhere.

File: server/schema/mutations.js (hunk 2 above)
Comment (kpjackson90) [reply server/schema/mutations.js]: @nashawn-griffith where did you destructure it?

Comment (nashawn-griffith) [reply server/schema/mutations.js]: {user} - this is where it was destructured

Comment (kpjackson90) [reply server/schema/mutations.js]: ohok, I see what going on.
